File: 169-majority-element/169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        map<int,int> occurenceCount;
        int n = nums.size();
        for(auto num: nums){
            occurenceCount[num]++;
        }
        
        int maxNum,maxCount=0;
        
        for(auto i:occurenceCount){
            if(i.second>maxCount){
                maxCount = i.second;
                maxNum = i.first;
            }
        }
        
        return maxNum;
    }
};

File: 169-majority-element/169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        ans = nums[0]
        count = 1
        
        for i in range(1,len(nums)):
            cur = nums[i]
            if count==0:
                ans = cur
            if ans==cur:
                count+=1
            else:
                count-=1
        return ans